### PR TITLE
Implement expand

### LINKED ISSUE #224: repro example does not work
```r
library(data.table)
library(tidyverse)
library(dplyr)

mtcars2 <- lazy_dt(mtcars)
mtcars2 %>% 
  filter(wt < 5) %>% 
  mutate(l100k = 235.21 / mpg) %>% # liters / 100 km
  group_by(cyl) %>% 
  summarise(l100k = mean(l100k))

```
gives:
Error in env_get_list(ns_env("data.table"), dt_funs) : 
  argument "default" is missing, with no default

### PR DESCRIPTION
**Pull request by markfairbanks**
Implements `expand()` for usage on lazy data.tables.

A couple other small adjustments:
* Removed a pipe from the internal `slice_min_max()` helper that was causing a note from `devtools::check()`.
* Bumped minimum data.table version to 1.13.0 to fix #224

### CODE REVIEW

File: R/step-subset-expand.R [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  }
+  names(dots) <- vctrs::vec_as_names(names(dots), repair = .name_repair)
+
+  on <- names(dots)
+
+  out <- distinct(data, !!!syms(data$groups), !!!dots)
+
+  cj <- expr(CJ(!!!syms(on), unique = TRUE))
+
+  if (length(data$groups) == 0) {
```
Comment (hadley) [inline R/step-subset-expand.R:66]:
```suggestion
  on <- names(dots)
  cj <- expr(CJ(!!!syms(on), unique = TRUE))

  out <- distinct(data, !!!syms(data$groups), !!!dots)
  if (length(data$groups) == 0) {
```

File: R/step-subset-slice.R [hunk 2]
```diff
@@ -157,8 +157,9 @@ slice_min_max <- function(.data, order_by, decreasing, ..., n, prop, with_ties =
     prop = expr(!!smaller_ranks(!!order_by, !!size$prop * .N, ties.method = ties.method))
   )
 
-  step_subset_i(.data, i) %>%
-    arrange(!!order_by, .by_group = TRUE)
+  out <- step_subset_i(.data, i)
+
+  arrange(out, !!order_by, .by_group = TRUE)
```
Comment (hadley) [inline R/step-subset-slice.R:162]:
```suggestion
  out <- step_subset_i(.data, i)
  arrange(out, !!order_by, .by_group = TRUE)
```

File: R/step-subset.R [hunk 3]
```diff
@@ -117,7 +119,11 @@ dt_call.dtplyr_step_subset <- function(x, needs_copy = x$needs_copy) {
 
   if (length(x$on) > 0) {
     out$on <- call2(".", !!!syms(x$on))
-    out$allow.cartesian <- TRUE
+    if (is.null(x$allow_cartesian)) {
+      out$allow.cartesian <- TRUE
+    } else {
+      out$allow.cartesian <- parent$allow_cartesian
+    }
```
Comment (hadley) [inline R/step-subset.R:126]:
```suggestion
    out$allow.cartesian <- parent$allow_cartesian %||% TRUE
```
?

**Review by hadley — COMMENTED** [review]
Looks good! Do you want to do `complete()` while you're in here?

File: R/step-subset.R (hunk 3 above)
Comment (markfairbanks) [reply R/step-subset.R]: I ended up adding an explicit `allow_cartesian = TRUE` to `step_subset_on()` and it made this part much more straightforward

**Comment by markfairbanks** [comment]
> Looks good! Do you want to do complete() while you're in here?

Yep, sounds good to me. I'll leave a comment when it's all ready.

**Comment by markfairbanks** [comment]
All set for review

**Comment by hadley** [comment]
Perfect, thanks!

**Pull request merged**